src/gui/processing/qgsprocessingwidgetwrapperimpl.cpp

```diff
@@ -613,6 +614,65 @@ QgsAbstractProcessingParameterWidgetWrapper *QgsProcessingAuthConfigWidgetWrappe
 // QgsProcessingNumericWidgetWrapper
 //
 
+QgsProcessingNumberParameterDefinitionWidget::QgsProcessingNumberParameterDefinitionWidget( QgsProcessingContext &context, const QgsProcessingParameterWidgetContext &widgetContext, const QgsProcessingParameterDefinition *definition, const QgsProcessingAlgorithm *algorithm, QWidget *parent )
+  : QgsProcessingAbstractParameterDefinitionWidget( context, widgetContext, definition, algorithm, parent )
+{
+  QVBoxLayout *vlayout = new QVBoxLayout();
+  vlayout->setMargin( 0 );
+  vlayout->setContentsMargins( 0, 0, 0, 0 );
+
+  vlayout->addWidget( new QLabel( tr( "Number type" ) ) );
+
+  mTypeComboBox = new QComboBox();
+  mTypeComboBox->addItem( tr( "Float" ), QgsProcessingParameterNumber::Double );
+  mTypeComboBox->addItem( tr( "Integer" ), QgsProcessingParameterNumber::Integer );
+  vlayout->addWidget( mTypeComboBox );
+
+  vlayout->addWidget( new QLabel( tr( "Minimum value" ) ) );
+  mMinLineEdit = new QLineEdit();
+  vlayout->addWidget( mMinLineEdit );
+
+  vlayout->addWidget( new QLabel( tr( "Maximum value" ) ) );
+  mMaxLineEdit = new QLineEdit();
+  vlayout->addWidget( mMaxLineEdit );
+
+  vlayout->addWidget( new QLabel( tr( "Default value" ) ) );
+  mDefaultLineEdit = new QLineEdit();
+  vlayout->addWidget( mDefaultLineEdit );
+
+  if ( const QgsProcessingParameterNumber *numberParam = dynamic_cast<const QgsProcessingParameterNumber *>( definition ) )
+  {
+    mTypeComboBox->setCurrentIndex( mTypeComboBox->findData( numberParam->dataType() ) );
+    mMinLineEdit->setText( QString::number( numberParam->minimum() ) );
+    mMaxLineEdit->setText( QString::number( numberParam->maximum() ) );
+    mDefaultLineEdit->setText( numberParam->defaultValue().toString() );
+  }
+
+  setLayout( vlayout );
+}
+
+QgsProcessingParameterDefinition *QgsProcessingNumberParameterDefinitionWidget::createParameter( const QString &name, const QString &description, QgsProcessingParameterDefinition::Flags flags ) const
+{
+  QgsProcessingParameterNumber::Type dataType = static_cast< QgsProcessingParameterNumber::Type >( mTypeComboBox->currentData().toInt() );
+  auto param = qgis::make_unique< QgsProcessingParameterNumber >( name, description, dataType, mDefaultLineEdit->text() );
```

Comment:
We should convert mDefaultLineEdit->text() to a double if possible

Reply:
Fixed